providers/src/airflow/providers/openlineage/utils/utils.py

```diff
@@ -262,9 +262,31 @@ class DagInfo(InfoJsonEncodable):
         "start_date",
         "tags",
     ]
-    casts = {"timetable": lambda dag: dag.timetable.serialize() if getattr(dag, "timetable", None) else None}
+    casts = {"timetable": lambda dag: DagInfo.serialize_timetable(dag)}
     renames = {"_dag_id": "dag_id"}
 
+    @classmethod
+    def serialize_timetable(cls, dag):
+        serialized = dag.timetable.serialize()
+        if serialized != {} and serialized is not None:
+            return serialized
+        if (
+            hasattr(dag, "dataset_triggers")
+            and isinstance(dag.dataset_triggers, list)
+            and len(dag.dataset_triggers) > 0
+        ):
+            triggers = dag.dataset_triggers
+            return {
+                "dataset_condition": {
```

Comment:
I think we probably would need to make it asset compatible?

Reply:
or is it used only in 2.8 and 2.9

Reply:
@Lee-W Yes, this is only for 2.8 and 2.9.